lib/semantics/resolve-names.cc

```diff
@@ -3568,6 +3571,78 @@ bool DeclarationVisitor::Pre(const parser::StructureConstructor &x) {
   return false;
 }
 
+bool DeclarationVisitor::Pre(const parser::BasedPointerStmt &x) {
+  for (const parser::BasedPointer &bp : x.v) {
+    const parser::ObjectName &pointerName{std::get<0>(bp.t)};
+    const parser::ObjectName &pointeeName{std::get<1>(bp.t)};
+    auto *pointer{FindSymbol(pointerName)};
+    if (!pointer) {
+      pointer = &MakeSymbol(pointerName, ObjectEntityDetails{});
+    } else if (pointer->has<UseDetails>()) {
+      Say(pointerName,
+          "'%s' cannot be a Cray pointer as it is use-associated"_err_en_US);
```

Comment:
You're right, we can and should allow it.

It's confusing because the pointer statement is both a declaration and an association. If the pointer is already declared and of the right type we should be able to associate a pointee with it, even if it is use-associated.